ultimatepython/advanced/file_handling.py

```diff
@@ -0,0 +1,49 @@
+"""
+File handling is a fundamental concept in Python that involves opening, reading, writing, and appending to files.
+This code example demonstrates the basics of file handling in Python.
+Python provides various ways to work with files.
+We can use the built-in 'open' function to open files in different modes like reading ('r'), writing ('w'), and appending ('a').
+"""
+
+
+# Open a file for reading
+def read_file(filename):
+    with open(filename, "r") as file:
+        content = file.read()
+        return content
+
+
+# Open a file for writing
+def write_file(filename, content):
+    with open(filename, "w") as file:
+        file.write(content)
+    return f"Content written to '{filename}'."
+
+
+# Open a file for appending
+def append_file(filename, content):
+    with open(filename, "a") as file:
+        file.write(content)
+    return f"Content appended to '{filename}'."
+
+
+def main():
+    # Example of writing to a file
+    result = write_file("sample.txt", "This is a test.")
```

Comment:
Can we store sample.txt as a global variable in this module? Maybe `_TARGET_FILE` as an example